File: php/features.php
<?php
declare(strict_types=1);

// WebsiteAnalysisApis2 SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class WebsiteAnalysisApis2Features
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new WebsiteAnalysisApis2BaseFeature();
            case "test":
                return new WebsiteAnalysisApis2TestFeature();
            default:
                return new WebsiteAnalysisApis2BaseFeature();
        }
    }
}

File: py/features.py
# WebsiteAnalysisApis2 SDK feature factory

from feature.base_feature import WebsiteAnalysisApis2BaseFeature
from feature.test_feature import WebsiteAnalysisApis2TestFeature


def _make_feature(name):
    features = {
        "base": lambda: WebsiteAnalysisApis2BaseFeature(),
        "test": lambda: WebsiteAnalysisApis2TestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
